nvtabular/loader/backend.py

```diff
@@ -243,6 +243,20 @@ def _gather_indices_for_dev(self, dev):
         start = self.global_rank * per_worker
         return self.indices[start : start + per_worker].tolist()
 
+    def _generate_local_seed(self):
+        random_state = cp.random.get_random_state()
+        seeds = random_state.tomaxint(size=self.global_size)
+        local_seed = seeds[self.global_rank]
+        cp.random.seed(local_seed.get())
+
+    def _shuffle_indices(self):
+        self._generate_local_seed()
```

Comment:
This might be a silly question - but why do we need to generate the seed before and after shuffling? Could we just generate this once on dataloader init?